com.unity.ml-agents.extensions/Runtime/Sensors/GridSensorComponent.cs

```diff
@@ -0,0 +1,326 @@
+        /// will be resized to double its current size.  The hard coded absolute size is 500.
+        /// Note that changing this after the sensor is created has no effect.
+        /// </summary>
+        public int InitialColliderBufferSize
+        {
+            get { return m_InitialColliderBufferSize; }
+            set { m_InitialColliderBufferSize = value; }
+        }
+
+        [HideInInspector, SerializeField]
+        Color[] m_DebugColors;
+        /// <summary>
+        /// Array of Colors used for the grid gizmos.
+        /// </summary>
+        public Color[] DebugColors
+        {
+            get { return m_DebugColors; }
+            set { m_DebugColors = value; }
+        }
+
+        [HideInInspector, SerializeField]
+        float m_GizmoYOffset = 0f;
+        /// <summary>
+        /// The height of the gizmos grid.
+        /// </summary>
+        public float GizmoYOffset
+        {
+            get { return m_GizmoYOffset; }
+            set { m_GizmoYOffset = value; }
+        }
+
+        [HideInInspector, SerializeField]
+        bool m_ShowGizmos = false;
+        /// <summary>
+        /// Whether to show gizmos or not.
+        /// </summary>
+        public bool ShowGizmos
+        {
+            get { return m_ShowGizmos; }
+            set { m_ShowGizmos = value; }
+        }
+
+        [HideInInspector, SerializeField]
+        SensorCompressionType m_CompressionType = SensorCompressionType.PNG;
+        /// <summary>
+        /// The compression type to use for the sensor.
+        /// </summary>
+        public SensorCompressionType CompressionType
+        {
+            get { return m_CompressionType; }
+            set { m_CompressionType = value; UpdateSensor(); }
+        }
+
+        [HideInInspector, SerializeField]
+        [Range(1, 50)]
+        [Tooltip("Number of camera frames that will be stacked before being fed to the neural network.")]
+        int m_ObservationStacks = 1;
+        /// <summary>
+        /// Whether to stack previous observations. Using 1 means no previous observations.
+        /// Note that changing this after the sensor is created has no effect.
+        /// </summary>
+        public int ObservationStacks
+        {
+            get { return m_ObservationStacks; }
+            set { m_ObservationStacks = value; }
+        }
+
+        /// <inheritdoc/>
+        public override ISensor[] CreateSensors()
+        {
+            m_Sensor = new GridSensor(
+                m_SensorName,
+                m_CellScaleX,
+                m_CellScaleY,
+                m_CellScaleZ,
+                m_GridNumSideX,
+                m_GridNumSideZ,
+                m_RotateWithAgent,
+                m_ChannelDepth,
+                m_DetectableObjects,
+                m_ObserveMask,
+                m_DepthType,
+                RootReference,
+                m_MaxColliderBufferSize,
+                m_InitialColliderBufferSize,
+                m_DebugColors,
```

Comment:
Can m_DebugColors and m_ShowGizmos be removed from GridSensor and kept to the component? (OK if that comes in a later PR).

Reply:
yes I've removed logic using m_DebugColors from the sensor, just need to remove it from constructor

The sensor needs m_ShowGizmos though to decide whether it has to fill the color indices buffer.